src/string.js

```diff
@@ -35,3 +36,17 @@ export let autoLabelOption = a => ({
   label: a.label || autoLabel(a.value || a),
 })
 export let autoLabelOptions = _.map(autoLabelOption)
+
+export let toSentenceWith = _.curry((initial, last, join, array) =>
+  _.flow(
+    mapIndexed((x, i) => [
+      x,
+      i === array.length - 1 ? null : i === array.length - 2 ? last : initial,
+    ]),
+    _.flatten,
+    _.compact,
```

Comment:
it seems like this might be cleaner with direct concatenation instead of flatten and compact
```js
      (x, i) =>
        x +
        (i < array.length - 1
          ? i < array.length - 2
            ? separator
            : lastSeparator
          : '')
```

Reply:
It's not what I'm intending to do with this function! See my comment here: https://github.com/smartprocure/futil-js/pull/246#discussion_r222250673

Reply:
I'll paste it here, since it got minified by some code I pushed:

Join could be F.arrayToObject. For React components, join could be a Component, like children => <div>{children}</div> :) this way we can wrap each matching element in any style and not have to force ourselves to have a string :) it could also be used for tag components or any kind of component that could resemble a sentence with connectors!!

Reply:
I really want joinWith to not necessarily output a string. Please let's make it more variable. I want to use it with components. Please 🙏 forcing toSentenceWith to be a string feels wrong to me, it's not generic and it's way too limited for the real necessity behind the concept of sentences.